cdap-kubernetes/pom.xml

```diff
@@ -86,6 +86,12 @@
         </exclusion>
       </exclusions>
     </dependency>
+    <dependency>
+      <groupId>io.cdap.cdap</groupId>
+      <artifactId>cdap-common</artifactId>
```

Comment:
Why? the kube module shouldn't have dependency on common.

Reply:
RemoteClient (i.e. libs to build http request and make REST API calls) brings in the dependency. 

To avoid this dependency, would need to duplicate some code over. Let me know if you see any better alternative.

Reply:
First of all, the kube module (or any master-spi extension for that matter), is designed with classloader isolation, so adding dependency on cdap-common means it will bring a different copy (and all of its transitive dependencies) into the extension directory. The cdap-common traditionally is a dump place for internal classes that needs to be shared among multiple modules, which makes it kind of big and full of weird dependencies.

If it is just for downloading a file from a http endpoint purpose, using the URLConnection directly doesn't really make the code more complicated (in fact might be easier).

Reply:
Reverted. 

Moved to MasterEnvironment. Tried to move to MasterEnvironmentContext, but no disocveryService there (which gets created after context creation)